telemetry/Tracks/class-tracks-client.php

```diff
@@ -0,0 +1,187 @@
+<?php
+/**
+ * Telemetry: Tracks Pixel class
+ *
+ * @package Automattic\VIP\Telemetry
+ */
+
+declare(strict_types=1);
+
+namespace Automattic\VIP\Telemetry;
+
+use WP_Error;
+
+/**
+ * Handles all operations related to the Tracks pixel.
+ */
+class Tracks_Client {
+	/**
+	 * Base URL of the Tracks pixel.
+	 */
+	protected const PIXEL_BASE_URL = 'https://pixel.wp.com/t.gif';
+
+	/**
+	 * Class singleton instance.
+	 *
+	 * @var ?Tracks_Client
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Events queued to be sent to the Tracks pixel.
+	 *
+	 * @var array<Tracks_Event>
+	 */
+	protected $events = array();
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		// Track events asynchronously (inject pixels in the footer).
+		add_action( 'admin_footer', array( $this, 'render_tracking_pixels' ) );
```

Comment:
Yes that is intentional. We're limiting this to WP admin for now. I'm also still trying to figure out a good way to make this part flexible so that where it renders the pixel is determined by the plugin implementation 🤔

Reply:
Why would it be consequential to the plugins?

Reply:
It'd be worth expanding this to the frontend as well imo. If we wanted, we could add a marker in the event to say when it's frontend or WP Admin and target the event recording based on that but I don't think it's worth it?